metricflow/model/transformations/dbt_to_metricflow.py

```diff
@@ -84,16 +95,21 @@ def _resolve_metric_model_ref(self, dbt_metric: DbtMetric) -> DbtModelNode:  # n
         else:
             ref_invalid_args(dbt_metric.name, ref_parts)
 
-        node = self.manifest.resolve_ref(
-            target_model_name=target_model,
-            target_model_package=target_package,
-            current_project=self.manifest.metadata.project_id,
-            node_package=dbt_metric.package_name,
-        )
-        assert isinstance(
-            node, DbtModelNode
-        ), f"Ref `{dbt_metric.model}` resolved to {node}, which is not of type `{DbtModelNode.__name__}`"
-        return node
+        hashed = hash((target_model, target_package, self.manifest.metadata.project_id, dbt_metric.package_name))
+        if hashed not in self._resolved_dbt_model_refs:
```

Comment:
Does this make a meaningful difference in the time this operation takes? If not it might be worth simplifying the logic and leaving this as an open todo after launch.

Reply:
Also, we might consider something like functools.cache for this sort of thing. https://docs.python.org/3/library/functools.html#functools.cache

Then you just do the body of the if statement in a function and return that result here, and all the memoization stuff just... happens. Makes it easier to switch to an LRU if we end up needing to evict values and stuff, just change out the decorator.

Reply:
I agree that we should investigate further as to whether this is actually necessary. Additionally, if dbt doesn't already do this caching for a given manifest, we could contribute upstream to make it start thus improving the world for everyone 🌎